src/Components/Application/Application.jsx

```diff
@@ -62,32 +62,49 @@ function Application({ applications, newReview, history, match, user }) {
 
   //Returns a review from DB if exists, otherwise null
   useEffect(() => {
+    let hasUnmounted = false;
+
     GET.getReviewAPI(user, appId).then((res) => {
       const reviewExists = res != null;
       if (reviewExists) {
         isRated.current = res.rating > -1;
       }
-      dispatchReviewUpdate({
-        type: LOAD_REVIEW,
-        review: reviewExists ? res : createReview(user, appId)
-      });
+
+      if (!hasUnmounted) {
+        dispatchReviewUpdate({
+          type: LOAD_REVIEW,
+          review: reviewExists ? res : createReview(user, appId)
+        });
+      }
     });
+
+    return () => {
+      hasUnmounted = true;
+    };
   }, [appId, user]);
 
   //Updates a review when any update happens from the user
   useEffect(() => {
+    let hasUnmounted = false;
+
     if (appId == null || user == null || review == null) {
       return;
     }
     UPDATE.updateReviewAPI(review).then((res) => {
       if (!isRated.current && review.rating > -1) {
         isRated.current = true;
-        newReview();
+        if (!hasUnmounted) {
+          newReview();
```

Comment:
I think it's fine to leave this one, as newReview is actually a dispatcher for the global redux state